src/libopenrave/kinbodygeometry.cpp

```diff
@@ -563,19 +563,33 @@ inline void LoadJsonValue(const rapidjson::Value& v, KinBody::GeometryInfo::Side
     if(v.IsObject()) {
         orjson::LoadJsonValueByKey(v, "transform", t.transf);
         orjson::LoadJsonValueByKey(v, "halfExtents", t.vExtents);
-        std::string type = "";
-        orjson::LoadJsonValueByKey(v, "type", type);
-        std::map<std::string, KinBody::GeometryInfo::SideWallType> sideWallTypeMapping = {
-            {"nx", KinBody::GeometryInfo::SideWallType::SWT_NX},
-            {"px", KinBody::GeometryInfo::SideWallType::SWT_PX},
-            {"ny", KinBody::GeometryInfo::SideWallType::SWT_NY},
-            {"py", KinBody::GeometryInfo::SideWallType::SWT_PY},
-        };
-
-        if (sideWallTypeMapping.find(type) == sideWallTypeMapping.end()) {
-            throw OPENRAVE_EXCEPTION_FORMAT(_("unrecognized sidewall type %s for loading from json"), type, ORE_InvalidArguments);
+        if(!v.HasMember("type")) {
+            throw OPENRAVE_EXCEPTION_FORMAT0(_("need sidewall type when loading from json"), ORE_InvalidArguments);
+        }
+        int sideWallType = 0;
+        if (v["type"].IsInt()) {
+            // backward compatibility, support enum sideWall type
+            orjson::LoadJsonValueByKey(v, "type", sideWallType);
+            if (!(sideWallType >= KinBody::GeometryInfo::SideWallType::SWT_First
+                && sideWallType <= KinBody::GeometryInfo::SideWallType::SWT_Last)) {
+                throw OPENRAVE_EXCEPTION_FORMAT(_("unrecognized sidewall type enum range %d for loading from json"), sideWallType, ORE_InvalidArguments);
+            }
+        }
+        else {
+            std::string type = "";
+            orjson::LoadJsonValueByKey(v, "type", type);
+            std::map<std::string, KinBody::GeometryInfo::SideWallType> sideWallTypeMapping = {
```

Comment:
@bucketzxm having std::map initialized everytime is slow for no reason. having a `switch` statement is better/faster.

Reply:
This is a string mapping .. which can't do `switch` i guess ?

Reply:
You can use `if` and `==`, probably still faster.